transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java

```diff
@@ -270,7 +270,12 @@ protected void doWrite(ChannelOutboundBuffer in) throws Exception {
             try {
                 // Check if sendmmsg(...) is supported which is only the case for GLIBC 2.14+
                 if (Native.IS_SUPPORTING_SENDMMSG && in.size() > 1) {
-                    NativeDatagramPacketArray array = NativeDatagramPacketArray.getInstance(in);
+                    NativeDatagramPacketArray array = ((EpollEventLoop) eventLoop()).cleanDatagramPacketArray();
+                    ChannelOutboundBuffer outboundBuffer = unsafe().outboundBuffer();
+                    if (outboundBuffer == null) {
+                        break;
+                    }
+                    outboundBuffer.forEachFlushedMessage(array);
```

Comment:
Where was this done before? Yeah... I don't understand how the code worked before this change; I don't see anything that adds to the `NativeDatagramPacketArray`.

Reply:
forEachFlushedMessage does

Reply:
This was done before in the getInstance call